dashpay/src/main/kotlin/org/dashevo/dashpay/BlockchainIdentity.kt

```diff
@@ -1074,15 +1075,31 @@ class BlockchainIdentity {
     }
 
     // Contact Requests
-    fun addContactToWallet(contactKeyChain: FriendKeyChain) {
+    fun addContactToWallet(contactKeyChain: FriendKeyChain, encryptionKey: KeyParameter? = null) {
         when (contactKeyChain.type) {
-            FriendKeyChain.KeyChainType.RECEIVING_CHAIN -> wallet!!.addReceivingFromFriendKeyChain(contactKeyChain)
+            FriendKeyChain.KeyChainType.RECEIVING_CHAIN -> {
+                Preconditions.checkArgument(wallet!!.isEncrypted == (encryptionKey != null))
+                if (wallet!!.isEncrypted) {
+                    val encryptedContactKeyChain = contactKeyChain.toEncrypted(wallet!!.keyCrypter, encryptionKey)
+                    wallet!!.addReceivingFromFriendKeyChain(encryptedContactKeyChain)
+                } else {
+                    wallet!!.addReceivingFromFriendKeyChain(contactKeyChain)
+                }
+            }
```

Comment:
In order to avoid the need of calling double-bang operator multiple times, you can use a scope function, for instance  `run`, like this:
```
wallet!!.run {
	Preconditions.checkArgument(isEncrypted == (encryptionKey != null))
	if (isEncrypted) {
		val encryptedContactKeyChain = contactKeyChain.toEncrypted(keyCrypter, encryptionKey)
		addReceivingFromFriendKeyChain(encryptedContactKeyChain)
	} else {
		addReceivingFromFriendKeyChain(contactKeyChain)
	}
}
```